prelude.dx

```diff
@@ -233,6 +237,35 @@ randIdx key =
   unif = rand key
   asidx $ floor $ unif * real (size @n)
 
+'min / max etc
+
+minby :: A a::Data. (a->Real)->a->a->a
+minby f x y = select ((f x) < (f y)) x y
+
+min :: Real -> Real -> Real
+min = minby id
+
+minimumby :: A p q::Data. (q->Real)->(p=>q)->q
+minimumby f xs = fold xs.(asidx 0) (for i. minby f xs.i)
+
+minimum :: A p. (p=>Real)->Real
+minimum = minimumby id
+
+negf :: (a->Real) -> (a->Real)
+negf f x = -1.0 * (f x)
+
+maxby :: A a::Data. (a->Real)->a->a->a
+maxby f = minby (negf f)
```

Comment:
This is calling out for a composition operator. I'm tempted to co-opt `o` for composition. Haskell uses `.` but we're already overloading that in a few places. Then we could write `neg o f` instead of defining a separate `negf`.

Reply:
Yes.

Julia uses `∘`, and endowers the REPL (and all text editors via plugins) with ability to type that as `\circ` <kbd>tab</kbd>

Could be interresting here.

Could go with something like `($)` or `$$`